### ISSUE TITLE
MSSQL database deployment failure

### ISSUE THREAD

**Issue by craddm** [alan-turing-institute/data-safe-haven#1580]
## :white_check_mark: Checklist

- [x] I have searched open and closed issues for duplicates.
- [x] This is a problem observed when **deploying** a Data Safe Haven.
- [x] I can reproduce this with the [latest version](https://github.com/alan-turing-institute/data-safe-haven/releases).
- [x] I have read through the [documentation](https://alan-turing-institute.github.io/data-safe-haven/).
- [x] This isn't an open-ended question (open a [discussion](https://github.com/alan-turing-institute/data-safe-haven/discussions) if it is).

## :computer: System information

- **Operating System:** MacOS
- **Data Safe Haven version:** v4.1.0

## :no_entry_sign: Describe the problem

During deployment, the MSSQL database deployment initially fails. Rerunning the deployment script appears to succeed, but in fact just skips the deployment of the database entirely. Smoke tests on the SRD subsequently fail.

## :deciduous_tree: Log messages

<details>
<summary>Relevant log messages</summary>

```none
[x] Failed to ensure that 'GREEN\SG T2MSRDS System Administrators' user exists on: 'MSSQL-T2MSRDS'!
Failed SQL command was: IF NOT EXISTS(SELECT * FROM sys.database_principals WHERE name = 'GREEN\SG T2MSRDS System Administrators') CREATE USER [GREEN\SG T2MSRDS System Administrators] FOR LOGIN [GREEN\SG T2MSRDS System Administrators];
Error message: A connection was successfully established with the server, but then an error occurred during the login process. (provider: SSL Provider, error: 0 - The certificate chain was issued by an authority that is not trusted.) System.Management.Automation.CmdletInvocationException: Incorrect syntax was encountered while parsing ''. ---> Microsoft.SqlTools.ServiceLayer.BatchParser.BatchParserException: Incorrect syntax was encountered while parsing ''.
   at Microsoft.SqlTools.ServiceLayer.BatchParser.Parser.RaiseError(ErrorCode errorCode, Token token, String message)
   at Microsoft.SqlTools.ServiceLayer.BatchParser.Parser.ExecuteBatch(Int32 repeatCount)
   at Microsoft.SqlTools.ServiceLayer.BatchParser.Parser.ParseLines()
   at Microsoft.SqlServer.Management.PowerShell.ExecutionProcessor.ExecuteTSql(String sqlCommand)
   --- End of inner exception stack trace ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Management.Automation.Internal.PipelineProcessor.SynchronousExecuteEnumerate(Object input)
   at System.Management.Automation.PipelineOps.InvokePipeline(Object input, Boolean ignoreInput, CommandParameterInternal[][] pipeElements, CommandBaseAst[] pipeElementAsts, CommandRedirection[][] commandRedirections, FunctionContext funcContext)
   at System.Management.Automation.Interpreter.ActionCallInstruction`6.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame) System.Management.Automation.CmdletInvocationException: The pipeline has been stopped. ---> System.Management.Automation.PipelineStoppedException: The pipeline has been stopped.
   at System.Management.Automation.MshCommandRuntime.ThrowTerminatingError(ErrorRecord errorRecord)
   at System.Management.Automation.Cmdlet.ThrowTerminatingError(ErrorRecord errorRecord)
   at Microsoft.SqlServer.Management.PowerShell.ExecutionProcessor.ExecuteTSql(String sqlCommand)
   at Microsoft.SqlServer.Management.PowerShell.GetScriptCommand.ProcessRecord()
   --- End of inner exception stack trace ---
Value[1]        : 
  Code          : ComponentStatus/StdErr/succeeded
  Level         : Info
  DisplayStatus : Provisioning succeeded
  Message       : Invoke-SqlCmd : Incorrect syntax was encountered while parsing ''.
At C:\Packages\Plugins\Microsoft.CPlat.Core.RunCommandWindows\1.1.15\Downloads\script3.ps1:97 char:9
+         Invoke-SqlCmd -ServerInstance $serverInstance -Credential $sq ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ParserError: (:) [Invoke-Sqlcmd], BatchParserException
    + FullyQualifiedErrorId : ExecutionFailureException,Microsoft.SqlServer.Management.PowerShell.GetScriptCommand
 
Status          : Succeeded
Capacity        : 0
Count           : 0
2023-08-23 10:28:01 [FAILURE]: [x] Remote script execution has failed. Please check the output above before re-running this script.
PS>TerminatingError(): "Remote script execution has failed. Please check the output above before re-running this script."
>> TerminatingError(): "Remote script execution has failed. Please check the output above before re-running this script."
>> TerminatingError(): "Remote script execution has failed. Please check the output above before re-running this script."
>> TerminatingError(): "Remote script execution has failed. Please check the output above before re-running this script."
Remote script execution has failed. Please check the output above before re-running this script.

Exception: Remote script execution has failed. Please check the output above before re-running this script.
```

</details>

## :recycle: To reproduce

Deploy an SRE that uses an MSSQL database.

**Comment by craddm** [comment]
Note: this is fixed already locally - this issue is for tracking

**Comment by craddm** [comment]
Closed by #1582
